src/rebar3_sheldon_ast.erl

```diff
@@ -0,0 +1,97 @@
+-module(rebar3_sheldon_ast).
+
+-export([spellcheck/1]).
+
+%% =============================================================================
+%% Spellcheck API
+%% =============================================================================
+
+-spec spellcheck([file:filename(), ...] | []) -> [maps:map(), ...] | [].
+spellcheck(Filenames) ->
+    spellcheck(Filenames, []).
+
+%% =============================================================================
+%% Internal functions
+%% =============================================================================
+
+-spec spellcheck([file:filename(), ...], list()) -> [maps:map(), ...] | [].
+spellcheck([], Acc) ->
+    Acc;
+spellcheck([Filename | T], Acc) ->
+    Comments = collect_comments(Filename),
+    Strings = collect_strings(Filename),
+    Result = lists:sort(Comments ++ Strings),
+    NewAcc =
+        lists:foldl(fun({Line, Type, Str}, Res) ->
+                       case sheldon:check(Str) of
+                           ok ->
+                               Res;
+                           Warnings ->
+                               [#{filename => Filename,
+                                  line => Line,
+                                  type => Type,
+                                  string => Str,
+                                  reason => Warnings}
+                                | Res]
+                       end
+                    end,
+                    Acc,
+                    Result),
+    spellcheck(T, NewAcc).
+
+-spec collect_strings(string()) ->
+                         [{erl_anno:anno(), binary | string | comment, string()}].
+collect_strings(Filename) ->
+    {ok, Bin} = file:read_file(Filename),
+    TrimBin = binary:replace(Bin, [<<"~p">>, <<"~n">>, <<"@">>], <<>>, [global]),
+    {ok, Tokens, _} = erl_scan:string(binary_to_list(TrimBin)),
+    collect_strings(Tokens, []).
+
+-spec collect_strings([term(), ...], list()) ->
+                         [{integer(), string | binary, string()}, ...] | [].
+collect_strings([], Acc) ->
+    Acc;
+collect_strings([{'<<', Line}, {string, Line, Str} | T], Acc) ->
+    case string:find(Str, "/*", leading) of
```

Comment:
This was done for skip in configs eg:
```erlang
{string,22,"src/*.erl"}

{string,22,"src/*/*.erl"}

{string,22,"include/*.hrl"}

{string,23,"src/*_ignore.erl"}
```
Maybe not a best solution but it was done for this case.

Reply:
Oh, no… don't do that… `/*` is too specific and you'll only be avoiding some very specific paths… and not avoiding stuff like `the_module.?rl`, for instance.
The actual issue here is that we will be running this on _every_ string and not just on "text" ones. And that's unavoidable.
Maybe we can add one or both of these options somewhere:
* `min_string_length :: pos_integer()`: With the minimal number of words in the string to be analyzable (so if you set it to 2 you avoid most configuration options which are one-word strings)
* `skip_if_special_chars :: boolean()`: So that if it's `true` and a word to be analyzed contains stuff that's not a _regular character_ (whatever that means), it's ignored.
But these two would work only if `"src/*_ignore.erl"` is considered a single word.

Reply:
Done